### PR TITLE
INS-17949 Add synchronous_mode_strict option for PostgreSQL clusters

### PR DESCRIPTION
**Pull request by alwyndavis**

### CODE REVIEW

**Review by michael-carter-instaclustr — APPROVED** [review]

File: acc_test/data/valid_postgresql_cluster_create.tf [hunk 1]
```diff
@@ -25,6 +25,7 @@ resource "instaclustr_cluster" "validPostgresql" {
       postgresql_node_count = 1,
       client_encryption = false,
       replication_mode = "SYNCHRONOUS",
+      synchronous_mode_strict = true
```
Comment (tharinduamila-insta) [inline acc_test/data/valid_postgresql_cluster_create.tf:28]: [MINOR]
Would be better to have as false which is the default for API.

File: docs/resources/cluster.md [hunk 2]
```diff
@@ -103,6 +103,7 @@ Property | Description | For Bundles | Default
 `zookeeper_node_count`|Indicates how many nodes are allocated to be Zookeeper nodes. For KAFKA bundle, if `dedicated_zookeeper` is false, then it indicates how many Kafka nodes also have ZooKeeper services in them. |Kafka, ZooKeeper
 `postgresql_node_count`|The number of nodes in a generated PostgreSQL cluster.|Postgresql|Required (Integers)
 `replication_mode` | The default replication behaviour for PostgreSQL cluster. See [Replication Mode](https://www.instaclustr.com/support/documentation/postgresql/options/replication-mode/) documentation for more details. Allowed values are `SYNCHRONOUS` and `ASYNCHRONOUS`. |Postgresql| `SYNCHRONOUS`
+`synchronous_mode_strict` | Whether writes should always require replication to at least one standby. See [Replication Mode](https://www.instaclustr.com/support/documentation/postgresql/options/synchronous-mode-strict/) documentation for more details. Allowed values are true or false. |Postgresql| false
```
Comment (tharinduamila-insta) [inline docs/resources/cluster.md]: Need to change the wording Replication Mode to Synchronous Mode Strict

File: acc_test/data/valid_postgresql_cluster_create.tf (hunk 1 above)
Comment (tharinduamila-insta) [reply acc_test/data/valid_postgresql_cluster_create.tf:28]: Wrong place for the comment. This is good.

File: examples/main.tf [hunk 3]
```diff
@@ -344,7 +344,8 @@ resource "instaclustr_cluster" "example-postgresql" {
     options = {
       postgresql_node_count = 1,
       client_encryption = true,
-      replication_mode = "SYNCHRONOUS"
+      replication_mode = "SYNCHRONOUS",
+      synchronous_mode_strict = true
```
Comment (tharinduamila-insta) [inline examples/main.tf:348]: [MINOR]
Would be better to have as false which is the default for API.

File: instaclustr/structs.go [hunk 4]
```diff
@@ -21,45 +21,46 @@ type OmitEmptyBool struct {
...
+	Truststore                    		string `json:"truststore,omitempty" mapstructure:"truststore,omitempty"`
+	RedisMasterNodes              		int    `json:"masterNodes,omitempty" mapstructure:"master_nodes,omitempty"`
+	RedisReplicaNodes             		int    `json:"replicaNodes,omitempty" mapstructure:"replica_nodes,omitempty"`
+	RedisPasswordAuth             		*bool  `json:"passwordAuth,omitempty" mapstructure:"password_auth,omitempty"`
+	DedicatedZookeeper            		*bool  `json:"dedicatedZookeeper,omitempty" mapstructure:"dedicated_zookeeper,omitempty"`
+	ZookeeperNodeSize             		string `json:"zookeeperNodeSize,omitempty" mapstructure:"zookeeper_node_size,omitempty"`
+	ZookeeperNodeCount            		int    `json:"zookeeperNodeCount,omitempty" mapstructure:"zookeeper_node_count,omitempty"`
+	PostgresqlNodeCount           		int    `json:"postgresqlNodeCount,omitempty" mapstructure:"postgresql_node_count,omitempty"`
+	PostgresqlReplicationMode     		string `json:"replicationMode,omitempty" mapstructure:"replication_mode,omitempty"`
+	PostgresqlSynchronousModeStrict     *bool `json:"synchronousModeStrict,omitempty" mapstructure:"synchronous_mode_strict,omitempty"`
```
Comment (tharinduamila-insta) [inline instaclustr/structs.go:63]: Is this misaligned or is just github showing it as such

**Review by tharinduamila-insta — CHANGES_REQUESTED** [review]

File: docs/resources/cluster.md (hunk 2 above)
Comment (alwyndavis) [reply docs/resources/cluster.md]: Tah, should be fixed now

File: examples/main.tf (hunk 3 above)
Comment (alwyndavis) [reply examples/main.tf:348]: Changed to false

File: instaclustr/structs.go (hunk 4 above)
Comment (alwyndavis) [reply instaclustr/structs.go:63]: I think Github?  In Intellij it looks aligned and I've run `terraform fmt`

**Review by tharinduamila-insta — APPROVED** [review]

**Review by tharinduamila-insta — APPROVED** [review]

**Review by michael-carter-instaclustr — APPROVED** [review]

**Pull request merged**